docs/react-types/ReactNode.md

```diff
@@ -0,0 +1,49 @@
+---
+title: ReactNode
+---
+
+`ReactNode` is a type that describes what React can render.
+
+## Parameters
+
+`ReactNode` does not take any parameters.
+
+## Usage
+
+### Typing `children`
+
+The most common use case for `ReactNode` is typing `children`.
+
+```tsx
+interface Props {
+  children?: ReactNode;
+}
+
+function Component({ children }: Props) {
+  return <>{children}</>;
```

Comment:
```suggestion
  return children;
```
Latest types and TS will support that and that's the proper default. Everything else are workarounds for bugs. We have those documented I think somewhere and should link to it.

Reply:
Definitely should be the default. The problem I get (with latest @types/react and typescript) is that `<Component>` implicitly gets `ReactNode` as return type, which gives type error when I try to render `<Component>` from another component: 

```tsx
import { ReactNode } from "react";

interface Props {
  children?: ReactNode;
}

function Component({ children }: Props) {
  return children;
}

function Example() {
  return <Component>Hello</Component>; // 'Component' cannot be used as a JSX component. Its return type 'ReactNode' is not a valid JSX element.
}
```

[TypeScript playground](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzgJQKYEMDGMByEAmqcAvnAGZQQhwBEUG2NA3AFAvAB2MqUZWRABUpgAzohZw4mABbAANvnocA-AC4UDXAVStibMgFcO2YBA5wAwlUgdUXABRIZ8xXZLqhEUQEpxk+jAGUObOCkq6+kYmZnAAogAe6OByqPa+CBJwAUHmADxW4GZ2MAB8ABKocnIQuQD0BTbFJUxwtbVwAOQNRVwdUugcHBDwAEZEBiKo+HDoYuhwAFIAygAaUtY9MAB0cACSMGLZwXAwAJ5gRB1oWFqEfcBiQ-DzAG7ocsDTy2uVqCDFWxYxCAA)

Should this not produce a type error in latest @types/react and typescript?

Reply:
Will be fixed either today or tomorrow: https://github.com/microsoft/TypeScript/issues/53031#issuecomment-1569015512 + https://github.com/DefinitelyTyped/DefinitelyTyped/pull/65135

Reply:
And now the TypeScript playground I linked just magically works. Amazing! Do you think we need to inform what version are required for this or should we should assume latest (as we actually state we do)?